### PR TITLE
Add C++ Units basic usage docs

### LINKED ISSUE #168: Add docs for C++ units library
Since a lot of the new features for 2020 use the units library, it would be useful to include a short guide on how to effectively use the library.

### PR DESCRIPTION
**Pull request by prateekma**
Closes #168

### CODE REVIEW

File: source/docs/software/basic-programming/cpp-units.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+   auto distance = 6_m;
+   auto time = 0.5_s;
+
+   units::meter_per_second_t speed = distance / time; // speed is 12_mps
+
+.. note:: When the calculation is made, ``auto`` is not used to ensure the validity of the calculation. When we divide distance by time, we want to make sure that a speed object (i.e. ``units::meter_per_second_t``) is being created. If the rvalue is not of type ``meter_per_second_t``, the compiler will throw an error. If ``auto`` is used, this check will not be made.
+
+Removing the unit wrapper
+~~~~~~~~~~~~~~~~~~~~~~~~~
+In order to expose the underlying type of the unit object, the ``to<..>()`` method can be used, where the template argument is the underlying type.
```
Comment (calcmogul) [inline source/docs/software/basic-programming/cpp-units.rst]: Remove "in order". It isn't needed grammatically and reduces wordiness.

**Comment by calcmogul** [comment]
For what it's worth, Wikipedia only caplitalizes the first word of sections rather than every word. :man_shrugging:

**Comment by prateekma** [comment]
LinkCheck fail is from some other article and not from any file in this PR.

**Comment by sciencewhiz** [comment]
I think it would be useful to show an example with a wpilib method that takes units

**Comment by prateekma** [comment]
@sciencewhiz Added a short example from sampling a trajectory.

**Pull request merged**